### PR TITLE
Add a passing test suite for POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by stevenwu99**
Please Task 7 for  POST /api/articles/:article_id/comments. Please check

### CODE REVIEW

File: controller/app.controller.js [hunk 1]
```diff
@@ -44,3 +44,18 @@ exports.getCommentsByArticleId = (req,res,next) => {
...
+   const article_id = req.params.article_id
+   const newComment = req.body;
+   const promises = [AddCommentByArticleId(article_id,newComment)];
+   if (article_id ) {
+       promises.push(checkExists('articles','article_id',article_id));
+   }
+   Promise.all(promises)
+   .then ((resolvedPromises) => {
+      const comments = resolvedPromises[0]
+      res.status(201).send({comments:comments[0]})
```
Comment (johnhutcheon) [inline controller/app.controller.js]: As we are only posting a new comment and not comments, could we change this key to 'comment' please?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -161,4 +161,43 @@ describe ('GET /api/articles/:article_id/comments',() => {
...
+    test ("400: should return an error if invalid article_id", () => {
+        return request(app)
+            .get("/api/articles/nonsense/comments")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Bad request");
+            });
+    });
+})
+
```
Comment (johnhutcheon) [inline __tests__/app.test.js:256]: The tests you’ve done so far are great, but there’s a few others we need to consider:

- A test that ignores any unnecessary properties if we send a comment with something else as well as a username and body
- An error test for missing required fields, e.g. no username or body properties sent in the comment
- An error test for if the username we send doesn’t exist

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Steven, nice work! Could you just make the amendments as per my comments, then resubmit please? Thanks! 🙂

File: controller/app.controller.js (hunk 1 above)
Comment (stevenwu99) [reply controller/app.controller.js]: done

File: __tests__/app.test.js (hunk 2 above)
Comment (stevenwu99) [reply __tests__/app.test.js:256]: You mean we do not need to test this one ,right?

A test that ignores any unnecessary properties if we send a comment with something else as well as a username and body

Comment (johnhutcheon) [reply __tests__/app.test.js:256]: Hey Steven, I mean you just need a test if the comment you sent was something like this:
{username: 'butter_bridge' , body: 'Very Good', bananas: 'bananas'}
It should ignore the extra property(bananas) and still post successfully!
Hope this makes sense!

Comment (stevenwu99) [reply __tests__/app.test.js:256]: thank you. complete your comments

**Review by johnhutcheon — APPROVED** [review]
Great work Steven, thanks for making the changes! 😄

**Pull request merged**